livedata/src/main/java/io/getstream/chat/android/livedata/EventBatchUpdate.kt

```diff
@@ -0,0 +1,113 @@
+package io.getstream.chat.android.livedata
+
+import io.getstream.chat.android.client.models.Channel
+import io.getstream.chat.android.client.models.Message
+import io.getstream.chat.android.client.models.User
+import io.getstream.chat.android.livedata.entity.ChannelEntity
+import io.getstream.chat.android.livedata.entity.MessageEntity
+import io.getstream.chat.android.livedata.extensions.users
+
+/**
+ * EventBatchUpdate helps you efficiently implement a 4 step batch update process
+ * It updates multiple messages, users and channels at once.
+ *
+ * val batchBuilder = EventBatchUpdate.Builder()
+ *
+ * as a first step specify which channels and messages to fetch
+ * batchBuilder.addToFetchChannels()
+ * batchBuilder.addToFetchMessages()
+ *
+ * as a second step, load the required data for batch updating using
+ * val batch = batchBuilder.build(domainImpl)
+ *
+ * third, add the required updates via
+ * batch.addUser, addChannel and addMessage methods
+ *
+ * fourth, execute the batch using
+ * batch.execute()
+ */
+internal class EventBatchUpdate private constructor (private val domainImpl: ChatDomainImpl, private val channelMap: Map<String, ChannelEntity>, private val messageMap: Map<String, MessageEntity>) {
+    val users: MutableMap<String, User> = mutableMapOf()
+    val channels: MutableMap<String, ChannelEntity> = mutableMapOf()
+    val messages: MutableMap<String, MessageEntity> = mutableMapOf()
```

Comment:
We have multiple props there having the same type, e.g.:
```kotlin
val messages: MutableMap<String, MessageEntity>
val messageMap: Map<String, MessageEntity>
```
It would be great to give them more meaningful names.